tools/cp-utility/src/main.rs

```diff
@@ -69,10 +82,51 @@ fn do_copy(operation: CopyOperation) -> io::Result<()> {
     match operation.copy_type {
         CopyType::Archive => copy_archive(&operation.source, &operation.destination)?,
         CopyType::SingleFile => fs::copy(&operation.source, &operation.destination).map(|_| ())?,
+        CopyType::Recursive => copy_recursive(&operation.source, &operation.destination)?,
     };
     Ok(())
 }
 
+fn copy_recursive(source: &Path, dest: &Path) -> io::Result<()> {
```

Comment:
What is the actual difference between this and `copy_archive`?  Could/should these be a single function with a parameter gating the difference?

Reply:
The main difference between these two functions is handling of the symbolic links. In `copy_recursive` it does not follow them by default https://www.gnu.org/software/coreutils/manual/html_node/cp-invocation.html and treat it as normal file.

on more difference that is currently not present/implemented between these two, is that copy_archive should store attributes of the file, where the copy_recursive will not. right now both copy_archive and copy_recursive are handling in same way without keeping any attributes of file. I am tending to keep these as seperate functions for copy_recursive and copy_archive function as it seems more clean to implement and unit tests . so copy_archive can be updated in future.